src/tree-view/__tests__/tree-view.test.tsx

```diff
@@ -0,0 +1,203 @@
+        description: 'Description 1.1',
+        hasActions: true,
+      },
+      {
+        id: '1.2',
+        title: 'Item 1.2',
+        description: 'Description 1.2',
+        items: [
+          {
+            id: '1.2.1',
+            title: 'Item 1.2.1',
+            description: 'Description 1.2.1',
+            hasActions: true,
+          },
+        ],
+      },
+      {
+        id: '1.3',
+        title: 'Item 1.3',
+        description: 'Description 1.3',
+        items: [
+          {
+            id: '1.3.1',
+            title: 'Item 1.3.1',
+            description: 'Description 1.3.1',
+          },
+        ],
+      },
+    ],
+  },
+  {
+    id: '2',
+    title: (
+      <Popover content="This is a popover" dismissButton={false}>
+        Item 2
+      </Popover>
+    ),
+    hasActions: true,
+    description: <Link href="#">Link in description</Link>,
+  },
+  {
+    id: '3',
+    title: 'Item 3',
+    description: 'Description 3',
+  },
+];
+
+const defaultProps: TreeViewProps<Item> = {
+  items: defaultData,
+  getItemId: item => item.id,
+  getItemChildren: item => item.items,
+  renderItem: item => ({
+    icon: <Icon name="folder" />,
+    content: item.title,
+    secondaryContent: item.description,
+    actions: item.hasActions ? defaultActions : undefined,
+    children: item.items,
+  }),
+};
+
+function renderTreeView(props: Partial<TreeViewProps<Item>> = {}) {
+  const { container, rerender } = render(<TreeView {...defaultProps} {...props} />);
+  const wrapper = createWrapper(container).findTreeView()!;
+  return { wrapper, rerender };
+}
+
+test('should render with only content', () => {
+  const { wrapper } = renderTreeView({ renderItem: item => ({ content: item.title }) });
+
+  const items = wrapper.findItems();
+  expect(items).toHaveLength(3);
+
+  items.forEach((item, index) => {
+    expect(item.findContent()?.getElement()).toHaveTextContent(`Item ${index + 1}`);
+    expect(item.findIcon()).toBeNull();
+    expect(item.findSecondaryContent()).toBeNull();
+    expect(item.findActions()).toBeNull();
+  });
+});
+
+test('should render with various slots', () => {
+  const { wrapper } = renderTreeView();
+
+  const items = wrapper.findItems();
+  expect(items).toHaveLength(3);
+
+  defaultData.forEach((data, index) => {
+    const item = items[index];
+
+    expect(item.findContent()!.getElement()).toHaveTextContent(`Item ${index + 1}`);
+    expect(item.findIcon()!.getElement()).toBeVisible();
+    if (data.description) {
+      expect(item.findSecondaryContent()!.getElement()).toBeVisible();
+
+      if (index !== 1) {
+        // index 1 (item 2)'s description is a link
+        expect(item.findSecondaryContent()!.getElement()).toHaveTextContent(`Description ${index + 1}`);
+      }
+    }
+    if (data.hasActions) {
+      expect(item.findActions()!.findButtonDropdown()!.getElement()).toBeVisible();
+    }
+  });
+});
+
+test('should render with other CDS components inside', () => {
```

Comment:
nit: I would remove the "CDS" from the test description. First of all, it might be unclear. Second, it is not really needed.